pkg/dockerregistry/server/manifestservice.go

```diff
@@ -21,6 +21,20 @@ import (
 	quotautil "github.com/openshift/origin/pkg/quota/util"
 )
 
+// ErrManifestBlobBadSize returned when a size of blob specified in the manifest
+// does not match the actual size. The docker/distribution does not check this
+// and therefore does not have a ready-made error for this.
+type ErrManifestBlobBadSize struct {
```

Comment:
Is it worth declaring and returning when turned into 500 error?

Reply:
@miminar This error will be part of `ErrManifestVerification`.

Reply:
@miminar It's ok for upstream to put unknown errors:

https://github.com/openshift/origin/blob/master/vendor/github.com/docker/distribution/registry/storage/signedmanifesthandler.go#L106

Reply:
@legionus for user it'll be "unknown error"